lib/espree.js

```diff
@@ -40,289 +151,377 @@ function convertAcornCommentToEsprimaComment(block, text, start, end, startLoc,
+                const loc = Parser.acorn.getLineInfo(this.input, pos);
 
-        parseTopLevel(node) {
-            if (this[STATE].impliedStrict) {
-                this.strict = true;
+                /** @type {EnhancedSyntaxError} */
+                const err = new SyntaxError(message);
+
+                err.index = pos;
+                err.lineNumber = loc.line;
+                err.column = loc.column + 1; // acorn uses 0-based columns
+                throw err;
             }
-            return super.parseTopLevel(node);
-        }
 
-        /**
-         * Overwrites the default raise method to throw Esprima-style errors.
-         * @param {int} pos The position of the error.
-         * @param {string} message The error message.
-         * @throws {SyntaxError} A syntax error.
-         * @returns {void}
-         */
-        raise(pos, message) {
-            const loc = Parser.acorn.getLineInfo(this.input, pos);
-            const err = new SyntaxError(message);
-
-            err.index = pos;
-            err.lineNumber = loc.line;
-            err.column = loc.column + 1; // acorn uses 0-based columns
-            throw err;
-        }
+            /**
+             * Overwrites the default raise method to throw Esprima-style errors.
+             * @param {int} pos The position of the error.
+             * @param {string} message The error message.
+             * @throws {EnhancedSyntaxError} A syntax error.
+             * @returns {void}
+             */
+            raiseRecoverable(pos, message) {
+                this.raise(pos, message);
+            }
 
-        /**
-         * Overwrites the default raise method to throw Esprima-style errors.
-         * @param {int} pos The position of the error.
-         * @param {string} message The error message.
-         * @throws {SyntaxError} A syntax error.
-         * @returns {void}
-         */
-        raiseRecoverable(pos, message) {
-            this.raise(pos, message);
-        }
+            /**
+             * Overwrites the default unexpected method to throw Esprima-style errors.
+             * @param {int} pos The position of the error.
+             * @throws {EnhancedSyntaxError} A syntax error.
+             * @returns {void}
+             */
+            unexpected(pos) {
+                let message = "Unexpected token";
 
-        /**
-         * Overwrites the default unexpected method to throw Esprima-style errors.
-         * @param {int} pos The position of the error.
-         * @throws {SyntaxError} A syntax error.
-         * @returns {void}
-         */
-        unexpected(pos) {
-            let message = "Unexpected token";
+                if (pos !== null && pos !== void 0) {
+                    this.pos = pos;
 
-            if (pos !== null && pos !== void 0) {
-                this.pos = pos;
+                    if (this.options.locations) {
+                        while (this.pos < /** @type {number} */ (this.lineStart)) {
```

Comment:
We have “int” defined. Should this be “int” instead of “number”?

Reply:
Yes, I think that is more helpful. Added.